xarray/core/computation.py

```diff
@@ -559,20 +559,29 @@ def func(*arrays):
                              'apply_ufunc: {}'.format(dask))
     result_data = func(*input_data)
 
-    if signature.num_outputs > 1:
-        output = []
-        for dims, data in zip(output_dims, result_data):
-            var = Variable(dims, data)
-            if keep_attrs and isinstance(args[0], Variable):
-                var.attrs.update(args[0].attrs)
-            output.append(var)
-        return tuple(output)
-    else:
-        dims, = output_dims
-        var = Variable(dims, result_data)
+    if signature.num_outputs == 1:
+        result_data = [result_data]
+
+    output = []
+    for dims, data in zip(output_dims, result_data):
```

Comment:
I think it would be nice if we raise an informative error message when the number of `output_dims` and `result_data` is different.
Currently this will silently drop some items if they have a different number of items.

Reply:
Good point -- done!